src/ParseUser.js

```diff
@@ -277,6 +277,13 @@ export default class ParseUser extends ParseObject {
    * @return {Boolean}
    */
   setUsername(username: string) {
+    // Strip anonymity, even we do not support anonymous user in js SDK, we may
+    // encoutner anonymous user created by android/ios in cloud code
+    var authData = this.get('authData');
+    if (typeof authData !== 'undefined' &&  authData.hasOwnProperty('anonymous')) {
```

Comment:
Looking at the existing style elsewhere in the SDK, this seems more consistent:

`if (authData && authData.hasOwnProperty('anonymous')) {`